pynamodb/attributes.py

```diff
@@ -387,27 +387,44 @@ def _get_attributes(cls):
                 cls._attributes[item] = instance
         return cls._attributes
 
+    def _get_in_memory_attributes(self):
+        try:
+            in_memory_attributes = getattr(self, 'in_memory_attributes')
```

Comment:
I'd go with `in_memory_attributes = self.in_memory_attributes` (which would raise `AttributeError` too), or `if hasattr(self, 'in_memory_attributes'): ...` to avoid raising the exception at all.